### DISCUSSION TITLE
Sorting search results by Title A-Z

### DISCUSSION THREAD

**Discussion by steveh2023** [microsoft-search/pnp-modern-search#3472] [Q&A] [answered]
Hi,

It was suggest to me to follow this info to solve this issue:

https://www.m365thinking.com/post/fix-one-of-the-problems-in-sharepoint-search-make-title-sortable

Which i did exactly but the sort is not a-z - the sort changed but not a-z or z-a, it just changed...

Does anyone know how i can get the results a-z please?

The link above is not overly complex and in our environment the details were replicated exactly and variations tried - nothing got the sort to be a-z...

Does anyone have any ideas on doing this please as the "standard" way is not working - or even any tips on troubleshooting this?

(Dev Tools then Network maybe?)

Thanks!

**Comment by kasperbolarsen** [comment]
I am the author of the blog post and I am curious why it doesn't work for you.
Can you share the Column definition for title, it should look similar to this:
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/3e6a6c60-599a-42d9-9199-4ccdbbe618c9)


Likewise I would like to see the sort setting, like this

![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/37352d7b-c3f7-4521-83ef-e1ae1fd22e91)

**Reply by steveh2023** [reply]
Hi,

None of the options to attach files work - so here is a screenshot of the settings:

https://ibb.co/Pcxn2wS

Do you know why the sort is still random/what is wrong/other things to try please?

Thanks!

**Reply by kasperbolarsen** [reply]
That looks OK. Could you verify that the query and sort properties in the API are correct. If you are not familiar with how to do that, this might be useful https://youtu.be/kPMSiRxK8H8?si=3eBaomcDpwax3KSX

**Reply by steveh2023** [reply]
It is working ok now - the fix was to map the refinablestring to the same list of properties that is mapped by default to Title...

**Reply by steveh2023** [reply]
Thanks for all your help - much appreciated!

**Reply by kasperbolarsen** [reply]
> It is working ok now - the fix was to map the refinablestring to the same list of properties that is mapped by default to Title...

Good to hear 😊 I believe I wrote that in the blog post? If it wasn't clear, I will update the post

**Reply by steveh2023** [reply]
It is not clear enough for someone new to editing the search schema - it was a colleague who knew what to map to - the list to map the string to is shown here: https://ibb.co/58h6Y45

**Reply by kasperbolarsen** [reply, marked as answer]
I have added the note in bold to the post, let's hope that is sufficient 

However, we can also see that the managed property "Title" is mapped to "Basic:#10, Office:#2, and Basic:#9, which we can use to fix this annoying issue.

We basically have to map one of the RefinableString properties to those crawled properties.

**NOTE: You should map ALL the crawled properties from Title to the RefinableString.**
